webv2/web.go

```diff
@@ -963,6 +1001,101 @@ func addIndexes(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(convm)
 }
 
+func getSourceDestinationSummary(w http.ResponseWriter, r *http.Request) {
+	sessionState := session.GetSessionState()
+	var sessionSummary sessionSummary
+	databaseType, err := helpers.GetSourceDatabaseFromDriver(sessionState.Driver)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error while getting source database: %v", err), http.StatusBadRequest)
+		return
+	}
+	sessionSummary.DatabaseType = databaseType
+	sessionSummary.ConnectionType = sessionState.SourceDBConnDetails.ConnectionType
+	sessionSummary.SourceTableCount = len(sessionState.Conv.SrcSchema)
+	sessionSummary.SpannerTableCount = len(sessionState.Conv.SpSchema)
+
+	sourceIndexCount, spannerIndexCount := 0, 0
+	for _, spannerSchema := range sessionState.Conv.SpSchema {
+		spannerIndexCount = spannerIndexCount + len(spannerSchema.Indexes)
+	}
+	for _, sourceSchema := range sessionState.Conv.SrcSchema {
+		sourceIndexCount = sourceIndexCount + len(sourceSchema.Indexes)
+	}
+	sessionSummary.SourceIndexCount = sourceIndexCount
+	sessionSummary.SpannerIndexCount = spannerIndexCount
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(sessionSummary)
+}
+
+func migrate(w http.ResponseWriter, r *http.Request) {
+
```

Comment:
Method is too long, consider refactoring to smaller units.

Reply:
done